subsys/bluetooth/services/fast_pair/fp_advertising.c

```diff
@@ -64,10 +64,19 @@ static int check_adv_config_range(struct bt_fast_pair_adv_config fp_adv_config)
 }
 
 static size_t bt_fast_pair_adv_data_size_non_discoverable(size_t account_key_cnt,
-						enum bt_fast_pair_adv_battery_mode adv_battery_mode)
+				const struct bt_fast_pair_not_disc_adv_info *not_disc_adv_info)
 {
 	size_t res = 0;
 
+	if (!IS_ENABLED(CONFIG_BT_FAST_PAIR_SUBSEQUENT_PAIRING) &&
+	    not_disc_adv_info->type == BT_FAST_PAIR_NOT_DISC_ADV_TYPE_SHOW_UI_IND) {
+		LOG_WRN("Cannot use not discoverable advertising with UI indications "
```

Comment:
`LOG_ERR` I think

Reply:
Consider moving it to `check_adv_config_range`, it seems to belong there

Reply:
we will discuss it offline. I think this function was only intended for checking the enum range

Reply:
I would say it could check for supported enum range aswell (it would be quite intuitive)

Reply:
Moved the validation logic to the `check_adv_config_range` and renamed the validation function to `check_adv_config`